hudi-common/src/main/java/org/apache/hudi/common/engine/TaskContextSupplier.java

```diff
@@ -35,4 +36,12 @@ public abstract class TaskContextSupplier implements Serializable {
   public abstract Supplier<Long> getAttemptIdSupplier();
 
   public abstract Option<String> getProperty(EngineProperty prop);
+
+  public Option<HoodieReaderContext> getReaderContext() {
```

Comment:
Can we make the task context supplier clean, does adding a new `#compactWithPartialUpdate` as interface solves the problem?

Reply:
The reader context object has to be created from the specific engine.

Reply:
If so, it should come from the `HoodieEngineContext` instead of `TaskContextSupplier`.

Reply:
Also, see if you can plug in the reader context from the engine-specific compaction handling as @danny0405 mentioned, to avoid new API in `HoodieEngineContext` or `TaskContextSupplier`.

Reply:
Yeah, maybe hudi table is a good place hold this context. Will check.